services/galley/src/Galley/API/LegalHold.hs

```diff
@@ -89,9 +90,21 @@ removeSettings :: UserId ::: TeamId ::: JSON -> Galley Response
 removeSettings (zusr ::: tid ::: _) = do
     membs <- Data.teamMembers tid
     void $ permissionCheck zusr ChangeLegalHoldTeamSettings membs
+    assertLegalHoldEnabled tid
 
+    let lhMembers = filter ((== UserLegalHoldEnabled) . view legalHoldStatus) membs
+    -- I picked this number by fair dice roll, feel free to change it :P
+    pooledMapConcurrentlyN_ 6 removeLHForUser lhMembers
```

Comment:
Let me know what you think of this; I figured it would be nice to have some concurrency, but I'm open to ideas.

Reply:
Looks good, but an unrelated concern would be that this request may take very long.  Should deletion lf devices be done in the background, and the request be answered immediately?  The client could make sure that it was successfull by checking whether LH is configured for the team?

No opinon on this.

Reply:
It definitely could take a while (hence trying to speed it up); the tricky bit is to ensure everything gets finished  correctly. Having the client need to check back later seems kind of like a cop-out, but for larger teams this could definitely be an issue.

Luckily, if it takes too long and "times out" it'll start from where it left off and will hopefully eventually finish 👍 

Open to ideas here, but maybe best to test it in the wild and see what happens?